### ISSUE TITLE
Suffix lock objects with term 'Lock'

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#510]
In case an object is used for synchronizing threads via lock statements, that object should be named with a suffix `Lock`. So it's easy to spot whether an object is used as synchronization mechanism for a lock.

Example:
```c#
lock (myLock) 
{
    ...
}
```

Note: `syncRoot` is also a valid name and should not trigger a violation.
